lib/internal/modules/run_main.js

```diff
@@ -148,7 +153,10 @@ function runEntryPointWithESMLoader(callback) {
  */
 function executeUserEntryPoint(main = process.argv[1]) {
   const resolvedMain = resolveMainPath(main);
-  const useESMLoader = shouldUseESMLoader(resolvedMain);
+  const tsEnabled = getOptionValue('--experimental-strip-types');
+  const useTSLoader = tsEnabled && shouldUseTSLoader(resolvedMain);
+  // If file is ts we should consider it as esm
```

Comment:
I think we should only treat it as esm if it is `.mts` or `package.json#type` is `"module"`.

Reply:
I mean stripping `.cts` to `.cjs`. Didn't mean to tranforms them in anyway.

Reply:
I see your point, I dont think it should be hard to do it, I will add a point to the tracking issue